speechbrain/utils/bleu.py

```diff
@@ -0,0 +1,151 @@
+import sacrebleu
+from speechbrain.dataio.dataio import merge_char, split_word
+from speechbrain.utils.metric_stats import MetricStats
+
+
+def merge_words(sequences):
+    results = []
+    for seq in sequences:
+        words = " ".join(seq)
+        results.append(words)
+    return results
+
+
+def detokenize_batch(detokenizer, sentences):
+    """
+    detokenizer: object
+        Detokenizer to be used
+    sentences: list
+        List of sentences to de detokenzied
+    """
+    detok_sentences = []
+    for sentence in sentences:
+        sentence = detokenizer.detokenize(sentence)
+        detok_sentences.append(sentence)
+
+    return detok_sentences
+
+
+class BLEUStats(MetricStats):
+    """A class for tracking BLEU.
+    Arguments
+    ---------
+    merge_tokens : bool
+        Whether to merge the successive tokens (used for e.g.,
+        creating words out of character tokens).
+        See ``speechbrain.dataio.dataio.merge_char``.
+    merge_words: bool
+        Whether to merge the successive tokens (used for e.g.,
+        creating words out of character tokens).
+    split_tokens : bool
+        Whether to split tokens (used for e.g. creating
+        characters out of word tokens).
+        See ``speechbrain.dataio.dataio.split_word``.
+    space_token : str
+        The character to use for boundaries. Used with ``merge_tokens``
+        this represents character to split on after merge.
+        Used with ``split_tokens`` the sequence is joined with
+        this token in between, and then the whole sequence is split.
+    Example
+    -------
+    >>> bleu = BLEUStats()
+    >>> i2l = {0: 'a', 1: 'b'}
+    >>> bleu.append(
+    ...     ids=['utterance1'],
+    ...     predict=[[0, 1, 1]],
+    ...     target=[[0, 1, 0]],
+    ...     ind2lab=lambda batch: [[i2l[int(x)] for x in seq] for seq in batch],
```

Comment:
This line might be confusing. Consider doing a line before and explaining it briefly.